services/search/generate_docs_emebddings.py

```diff
@@ -0,0 +1,125 @@
+import os
+import time
+import glob
+import argparse
+from dotenv import load_dotenv
+from langchain_text_splitters import RecursiveCharacterTextSplitter, MarkdownHeaderTextSplitter
+from pymilvus import FieldSchema, CollectionSchema, DataType, utility, Collection, connections, model
+
+load_dotenv()
+
+def read_md_files(directory):
+    md_files = glob.glob(f"{directory}/**/*.md", recursive=True)
+    docs = []
+    for file in md_files:
+        with open(file, "r", encoding="utf-8") as f:
+            docs.append((file, f.read()))  # Returning a tuple with file name and content
+    return docs
+
+def split_md_by_sections(file_name, content):
+    headers_to_split_on = [
+        ("##", "Header 2"),
+        ("###", "Header 3"),
+    ]
+    markdown_splitter = MarkdownHeaderTextSplitter(headers_to_split_on=headers_to_split_on, strip_headers=False)
+    md_header_splits = markdown_splitter.split_text(content)
+
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=512, chunk_overlap=64, length_function=len, is_separator_regex=False)
+    splits = text_splitter.split_documents(md_header_splits)
+
+    # Log the first line of each section and the file name to stdout
+    for section in splits:
+        first_line = section.page_content.splitlines()[0] if section.page_content.splitlines() else "[Empty Section]"
+        print(f"File: {file_name}, Section Start: {first_line}")
+
+    # Write the sections to disk
+    output_dir = "./tmp/split_sections"
+    os.makedirs(output_dir, exist_ok=True)
+    with open(os.path.join(output_dir, f"{os.path.basename(file_name)}_sections.md"), "w", encoding="utf-8") as out_file:
+        for section in splits:
+            out_file.write(f"## File: {file_name}\n")
```

Comment:
It was really hard to differentiate between the chunks, therefore I added this. Since this does not seem to be useful, I will just add a heading as `"## Section Start: " + "\n---\n"` for each chunk. It will be much more readable, I guess?